x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/policy_json_flyout.tsx

```diff
@@ -17,36 +18,108 @@ import {
   EuiSpacer,
   EuiText,
   EuiTitle,
+  EuiCallOut,
+  EuiLoadingSpinner,
 } from '@elastic/eui';
-import { Policy, PolicyFromES } from '../../../../../common/types';
-import { serializePolicy } from '../../../services/policies/policy_serialization';
+
+import { SerializedPolicy } from '../../../../../common/types';
+
+import { useFormContext, useFormData } from '../../../../shared_imports';
 
 interface Props {
+  legacyPolicy: SerializedPolicy;
   close: () => void;
-  policy: Policy;
-  existingPolicy?: PolicyFromES;
   policyName: string;
 }
 
 export const PolicyJsonFlyout: React.FunctionComponent<Props> = ({
-  close,
-  policy,
   policyName,
-  existingPolicy,
+  close,
+  legacyPolicy,
 }) => {
-  const { phases } = serializePolicy(policy, existingPolicy?.policy);
-  const json = JSON.stringify(
-    {
-      policy: {
-        phases,
-      },
-    },
-    null,
-    2
-  );
+  /**
+   * policy === undefined: we are checking validity
+   * policy === null: we have determined the policy is invalid
+   * policy === {@link SerializedPolicy} we have determined the policy is valid
+   */
+  const [policy, setPolicy] = useState<undefined | null | SerializedPolicy>(undefined);
+
+  const form = useFormContext();
```

Comment:
Minor nit, we don't need to validate on each form change if `isValid` is not undefined.

```js
const form = useFormContext();
const [formData, getFormData] = useFormData();
const { isValid, validate } = form;

useEffect(() => {
  (async function checkPolicy() {
    const isFormValid = isValid ?? (await validate());

    if (isFormValid) {
      const p = getFormData() as SerializedPolicy;
      setPolicy({
        ...legacyPolicy,
        phases: {
          ...legacyPolicy.phases,
          hot: p.phases.hot,
        },
      });
    } else {
      setPolicy(null);
    }
  })();
  // eslint-disable-next-line react-hooks/exhaustive-deps
}, [isValid, validate, legacyPolicy, formData]);
```

Reply:
Also in [`f5edadc` (#80842)](https://github.com/elastic/kibana/pull/80842/commits/f5edadc34964a45c2776aa5dcaf420bdace769de) I've fixed the problem that you faced and had to add the eslint override.

Reply:
Awesome, we should incorporate that into a following PR.